lib/SILOptimizer/SILCombiner/SILCombinerCastVisitors.cpp

```diff
@@ -452,6 +452,10 @@ SILCombiner::visitThickToObjCMetatypeInst(ThickToObjCMetatypeInst *TTOCMI) {
   if (CastOpt.optimizeMetatypeConversion(TTOCMI, MetatypeRepresentation::Thick))
     MadeChange = true;
 
+  if (auto OCTTMI = dyn_cast<ObjCToThickMetatypeInst>(TTOCMI->getOperand())) {
```

Comment:
Can you add a * here? That makes it clear to the reader that we have a pointer and no copy will occur without more work. It is a good habit to get into.

Reply:
I think you could also just return OCTTMI and SILCombine will handle the RAUW for you. One should only delete instructions like this inline generally.

Reply:
Added *

Reply:
I can't just return OCTTMI. I want to replace all the uses of `thick_to_objc_metatype` with the operand of `objc_to_thick_metatype`. If I return it as is, looks like SILCombine is trying to replace all uses of `thick_to_objc_metatype` with the result of `objc_to_thick_metatype `

Reply:
Updated with RAUW and eraseInstFromFunction